### ISSUE TITLE
Potential zcl_html_form improvements

### ISSUE THREAD

**Issue by sbcgua** [abapGit/abapGit#3559]
Followups and potential improvements from #3533

- [X] grouping of elements (for particularly large forms if even happen)

<details>
  <summary>experiments</summary>

```css
fieldset {
  border: none;
  border-top: 1px solid #ccc;
  margin-top: 8px;
}
fieldset legend {
  color: #ccc;
  font-size: small;
  text-transform: uppercase;
  padding-left: 1em;
  padding-right: 1em;
  margin-left: 0.3em;
}
```
![image](https://user-images.githubusercontent.com/15635498/85991142-ae466400-b9fb-11ea-9837-ea45bd9ad5fa.png)
</details>

- [ ] implementation of dropdowns (though better to avoid them and use radio buttons instead - [good talk](https://www.youtube.com/watch?v=hcYAHix-riY&feature=emb_logo) on this subject)
- [X] auto-generated form id of not passed (for double clicks to always work for side actions)
- [ ] maybe possibility to add several side actions
- [X] adjust markup: bring ul inside form, use div instead, adjust css respectively (last comment from PR)
- [X] move colors to theme
- [X] dev docs
- [x] integer field with min/max (from #3904)

**Comment by sbcgua** [comment]
I'll leave this for a while after #3582. The remainings are low prio nice-to-haves imho. So if needed, feel free to implement the remaining features or other features.

**Comment by sbcgua** [comment]
- [ ] Unified validation (bool and uppercase can be unified too)

**Comment by mbtools** [comment]
I don't think we need dropdowns or multiple side actions. Close?

**Comment by sbcgua** [comment]
Yeah. Seem to be fine so far.
